src/permutat.cc

```diff
@@ -1748,19 +1748,10 @@ static inline Obj RESTRICTED_PERM(Obj perm, Obj dom, Obj test)
       len = GET_LEN_RANGE(dom);
       p = GET_LOW_RANGE(dom);
       inc = GET_INC_RANGE(dom);
-      while (p<1) {
-        p+=inc;
-        len=-1;
-      }
-      i=p+(inc*len)-1;
-      while (i>deg) {
-        i-=inc;
-      }
-      p-=1;
-      i-=1;
-      while (p<=i) {
-        ptRest[p]=ptPerm[p];
-        p+=inc;
+      for ( i = p; i != p+inc*len; i += inc) {
+          if(i >= 1 && i <= deg) {
```

Comment:
You could combine these if you wanted to into something like:
```suggestion
      for ( i = Maximum(1,p); i != p+inc*len && i <= deg; i += inc) {
```
(Not sure how you do `Maximum`)

Reply:
Be aware that inc can be positive or negative, and p could start higher than deg (that's why I do i != p+inc*len, rather than the more usual i < p+inc*len).

Reply:
Oops, silly me, you're quite right. Please continuing ignoring my wrong suggestion ;-)